src/main/java/com/sprints/validations/ValidateDateFormat.java

```diff
@@ -0,0 +1,27 @@
+package com.sprints.validations;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.Optional;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class ValidateDateFormat {
```

Comment:
Are you transforming the Date that you obtain in the query param ?

Reply:
Yes, I transform the string to Date as it is stored in the Mongo database so then I can include it in the query